Description: Test against '127.0.0.1' instead of 'localhost', to work around a
 bug in gearmand which do not handle the case where a hostname resolves to
 multiple equal IP addresses.
Bug: https://bugs.launchpad.net/gearmand/+bug/918055
Author: Stig Sandbeck Mathisen <ssm@debian.org>
Last-Update: 2012-01-21
Forwarded: no

--- gearmand.orig/libtest/blobslap_worker.cc
+++ gearmand/libtest/blobslap_worker.cc
@@ -48,7 +48,7 @@
 private:
 public:
   BlobslapWorker(in_port_t port_arg) :
-    Server("localhost", port_arg)
+    Server("127.0.0.1", port_arg)
   { 
     set_pid_file();
   }
--- gearmand.orig/libtest/gearmand.cc
+++ gearmand/libtest/gearmand.cc
@@ -206,7 +206,7 @@
     arg_buffer << " -u root ";
   }
 
-  arg_buffer << " --listen=localhost ";
+  arg_buffer << " --listen=127.0.0.1 ";
 
   for (int x= 1 ; x < argc ; x++)
   {
--- gearmand.orig/libtest/memcached.cc
+++ gearmand/libtest/memcached.cc
@@ -298,7 +298,7 @@
     arg_buffer << " -u root ";
   }
 
-  arg_buffer << " -l localhost ";
+  arg_buffer << " -l 127.0.0.1 ";
   arg_buffer << " -m 128 ";
   arg_buffer << " -M ";
 
--- gearmand.orig/libtest/server_container.cc
+++ gearmand/libtest/server_container.cc
@@ -172,7 +172,7 @@
     {
       if (HAVE_LIBGEARMAN)
       {
-        server= build_gearmand("localhost", try_port);
+        server= build_gearmand("127.0.0.1", try_port);
       }
       else
       {
@@ -208,7 +208,7 @@
     {
       if (HAVE_LIBMEMCACHED)
       {
-        server= build_memcached_sasl("localhost", try_port, construct.username(), construct.password());
+        server= build_memcached_sasl("127.0.0.1", try_port, construct.username(), construct.password());
       }
       else
       {
@@ -226,7 +226,7 @@
     {
       if (HAVE_LIBMEMCACHED)
       {
-        server= build_memcached("localhost", try_port);
+        server= build_memcached("127.0.0.1", try_port);
       }
       else
       {
@@ -311,7 +311,7 @@
     {
       if (HAVE_LIBMEMCACHED)
       {
-        server= build_memcached_sasl_socket("localhost", try_port, username(), password());
+        server= build_memcached_sasl_socket("127.0.0.1", try_port, username(), password());
       }
       else
       {
@@ -329,7 +329,7 @@
     {
       if (HAVE_LIBMEMCACHED)
       {
-        server= build_memcached_socket("localhost", try_port);
+        server= build_memcached_socket("127.0.0.1", try_port);
       }
       else
       {
--- gearmand.orig/tests/drizzle_test.cc
+++ gearmand/tests/drizzle_test.cc
@@ -75,7 +75,7 @@
 static test_return_t gearmand_basic_option_test(void *)
 {
   const char *args[]= { "--check-args", 
-    "--libdrizzle-host=localhost",
+    "--libdrizzle-host=127.0.0.1",
     "--libdrizzle-port=90",
     "--libdrizzle-uds=tmp/foo.socket",
     "--libdrizzle-user=root",
--- gearmand.orig/tests/httpd_test.cc
+++ gearmand/tests/httpd_test.cc
@@ -61,7 +61,7 @@
 
 static test_return_t GET_TEST(void *)
 {
-  libtest::http::GET get("http://localhost:8090/");
+  libtest::http::GET get("http://127.0.0.1:8090/");
 
   test_compare(false, get.execute());
 
@@ -70,7 +70,7 @@
 
 static test_return_t HEAD_TEST(void *)
 {
-  libtest::http::HEAD head("http://localhost:8090/");
+  libtest::http::HEAD head("http://127.0.0.1:8090/");
 
   test_compare(false, head.execute());
 
--- gearmand.orig/tests/libgearman-1.0/client_test.cc
+++ gearmand/tests/libgearman-1.0/client_test.cc
@@ -293,7 +293,7 @@
 
   from_with_host= gearman_client_create(NULL);
   test_truth(from_with_host);
-  gearman_client_add_server(from_with_host, "localhost", 12345);
+  gearman_client_add_server(from_with_host, "127.0.0.1", 12345);
 
   client= gearman_client_clone(NULL, from_with_host);
   test_truth(client);
@@ -800,7 +800,7 @@
   test_truth(client_ptr);
 
   gearman_return_t rc;
-  rc= gearman_client_add_servers(&client, "localhost:4730,localhost");
+  rc= gearman_client_add_servers(&client, "127.0.0.1:4730,localhost");
   test_compare_got(GEARMAN_SUCCESS, rc, gearman_strerror(rc));
 
   rc= gearman_client_add_servers(&client, "old_jobserver:7003,broken:12345");
--- gearmand.orig/tests/memcached_test.cc
+++ gearmand/tests/memcached_test.cc
@@ -32,7 +32,7 @@
 {
   const char *args[]= { "--check-args",
     "--queue=libmemcached",
-    "--libmemcached-servers=localhost:12555", 
+    "--libmemcached-servers=127.0.0.1:12555", 
     0 };
 
   test_compare(EXIT_SUCCESS, exec_cmdline(gearmand_binary(), args, true));
@@ -43,7 +43,7 @@
 static test_return_t collection_init(void *object)
 {
   const char *argv[]= { "test_gearmand", 
-    "--libmemcached-servers=localhost:12555", 
+    "--libmemcached-servers=127.0.0.1:12555", 
     "--queue-type=libmemcached",
     0 };
 
--- gearmand.orig/tests/redis.cc
+++ gearmand/tests/redis.cc
@@ -33,7 +33,7 @@
 static test_return_t gearmand_basic_option_test(void *)
 {
   const char *args[]= { "--check-args", 
-    "--redis-server=localhost",
+    "--redis-server=127.0.0.1",
     "--redis-port=6379",
     0 };
 
